src/schematools/types.py

```diff
@@ -475,12 +503,12 @@ def get_through_tables_by_id(self) -> List[DatasetTableSchema]:
         return tables
 
     @property
-    def display_field(self):
+    def display_field(self) -> Optional[Any]:
         """Tell which fields can be used as display field."""
         return self["schema"].get("display", None)
 
-    def get_dataset_schema(self, dataset_id):
-        """Return another datasets"""
+    def get_dataset_schema(self, dataset_id: str) -> Optional[DatasetSchema]:
+        """Return another dataset"""
         return self._parent_schema.get_dataset_schema(dataset_id)
```

Comment:
How does this function work? Every `DatasetSchema` has a `_parent_schema` of type DatasetSchema that is required (it is not `Optional`)

Calling `get_dataset_schema` on a `DatasetSchema` instance recursively calls `get_dataset_schema` on its parent. How does it stop? In other words what is its stopping condition?

Reply:
That is not how it works. The `DatasetTableSchema` has a `_parent_schema` which is a `DatasetSchema`. So, there is no recursion involved.

Reply:
Ah, thanks for the clarification. I had missed that this was `DatasetTableSchema`'s `get_dataset_schema`.